File: src/dataset/go/difference_of_squares_Exercism_go_thinkverse/openAI/ChatGPT4_difference_of_squares_v2.go
package diffsquares

func SquareOfSum(n int) int {
	return (n * (n + 1) / 2) * (n * (n + 1) / 2)
}

func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

func Difference(n int) int {
	squareOfSum := (n * (n + 1) / 2)
	squareOfSum *= squareOfSum
	sumOfSquares := n * (n + 1) * (2*n + 1) / 6
	return squareOfSum - sumOfSquares
}

File: src/dataset/typescript/difference-of-squares_exercism-javascript-samajammin/openAI/ChatGPT4_difference-of-squares_v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Squares {
    constructor(num) {
        this.squareOfSum = this.calcSquareOfSum(num);
        this.sumOfSquares = this.calcSumOfSquares(num);
        this.difference = this.squareOfSum - this.sumOfSquares;
    }
    calcSquareOfSum(num) {
        const sum = (num * (num + 1)) / 2; // Sum of first N natural numbers
        return Math.pow(sum, 2);
    }
    calcSumOfSquares(num) {
        return (num * (num + 1) * (2 * num + 1)) / 6; // Sum of squares of first N natural numbers
    }
}
exports.default = Squares;
